tests/5.0/loop/test_loop_bind_device.F90

```diff
@@ -0,0 +1,246 @@
+    INTEGER,DIMENSION(OMPVV_NUM_TEAMS_DEVICE,N,N):: outData
+    INTEGER,DIMENSION(N,N):: x
+    INTEGER:: errors, num_teams, i, j, k
+
+    OMPVV_INFOMSG("test_loop_bind_thread_teams")
+
+    errors = 0
+    num_teams = -1
+
+    DO i = 1, N
+       DO j = 1, N
+          DO k = 1, OMPVV_NUM_TEAMS_DEVICE
+             outData(k,j,i) = 1
+          END DO
+       END DO
+       y(i) = i
+       z(i) = 2*i
+    END DO
+
+    !$omp target teams num_teams(OMPVV_NUM_TEAMS_DEVICE) thread_limit(OMPVV_NUM_THREADS_DEVICE) private(x) map(tofrom: outData, y, z, num_teams)
+    DO i = 1, N
+       DO j = 1, N
+          x(j,i) = 1
+       END DO
+    END DO
+    !$omp loop bind(thread)
+    DO i = 1, N
+       DO j = 1, N
+          x(j,i) = x(j,i) + y(i)*z(i)
+       END DO
+    END DO
+    !$omp end loop
+    !$omp parallel if(.FALSE.)
+    IF ( omp_get_team_num() .EQ. 0 ) THEN
+       num_teams = omp_get_num_teams()
+    END IF
+    !$omp end parallel
+    DO i = 1, N
+       DO j = 1, N
+          outData(omp_get_team_num()+1,j,i) = x(j,i)
+       END DO
+    END DO
+    !$omp end target teams
+
+    DO i = 1, N
+       DO j = 1, N
+          DO k = 1, num_teams
+             OMPVV_TEST_AND_SET_VERBOSE(errors, outData(k,j,i) .NE. (1 + y(i)*z(i)))
+          END DO
+       END DO
+    END DO
+
+    OMPVV_WARNING_IF(num_teams .EQ. 1, "Test ran with one team, so parallelism of loop construct can't be guaranteed.")
+    OMPVV_TEST_AND_SET_VERBOSE(errors, num_teams .LT. 1)
+    OMPVV_ERROR_IF(num_teams .LT. 1, "omp_get_num_teams() returned an invalid number of teams.")
+
+    test_bind_thread_teams = errors
+  END FUNCTION test_bind_thread_teams
+
+  INTEGER FUNCTION test_bind_thread_parallel()
+    INTEGER,DIMENSION(N):: y
+    INTEGER,DIMENSION(N):: z
+    INTEGER,DIMENSION(OMPVV_NUM_THREADS_HOST,N,N):: outData
+    INTEGER,DIMENSION(N,N):: x
+    INTEGER:: errors, num_threads, i, j, k
+
+    OMPVV_INFOMSG("test_loop_bind_thread_parallel")
+
+    errors = 0
+    num_threads = -1
+
+    DO i = 1, N
+       DO j = 1, N
+          DO k = 1, OMPVV_NUM_THREADS_HOST
+             outData(k,j,i) = 1
+          END DO
+       END DO
+       y(i) = i
+       z(i) = 2*i
+    END DO
+
+    !$omp target parallel shared(outData) num_threads(OMPVV_NUM_THREADS_DEVICE) private(x) map(tofrom: outData, y, z, num_threads)
+    DO i = 1, N
+       DO j = 1, N
+          x(j,i) = 1
+       END DO
+    END DO
+    !$omp loop bind(thread)
+    DO i = 1, N
+       DO j = 1, N
+          x(j,i) = x(j,i) + y(i)*z(i)
+       END DO
+    END DO
+    !$omp end loop
+    IF ( omp_get_thread_num() .EQ. 0 ) THEN
+       num_threads = omp_get_num_threads()
+    END IF
+    DO i = 1, N
+       DO j = 1, N
+          outData(omp_get_thread_num()+1,j,i) = x(j,i)
+       END DO
+    END DO
+    !$omp end target parallel
+
+    DO i = 1, N
+       DO j = 1, N
+          DO k = 1, num_threads
+             OMPVV_TEST_AND_SET_VERBOSE(errors, outData(k,j,i) .NE. (1 + y(i)*z(i)))
```

Comment:
I suggest changing `OMPVV_TEST_AND_SET_VERBOSE` for `OMPVV_TEST_AND_SET`

Reply:
Updated